libgstc/pygstc/gstc.py

```diff
@@ -215,44 +216,70 @@ def _send_cmd_line(self, cmd_line):
         result : dictionary
             Response from the IPC
         """
-
-        cmd = cmd_line[0]
         try:
+            cmd = cmd_line[0]
             jresult = self._ipc.send(cmd_line, timeout=self._timeout)
             result = json.loads(jresult)
-        except Exception as exception:
-            self._logger.error('%s error: %s' % (cmd,
-                                                 type(exception).__name__))
-            traceback.print_exc()
-            raise GstcError(type(exception).__name__)
-        if result['code'] != 0:
-            self._logger.error('%s error: %s' % (cmd,
-                                                 result['description']))
-            raise GstdError(result['description'])
-        return result
+            if result['code'] != GstcErrorCode.GSTC_OK.value:
+                self._logger.error('%s error: %s' % (cmd,
+                                                     result['description']))
+                raise GstdError(result['description'],
+                                result['code'])
+            return result
+        except ConnectionRefusedError as e:
+            raise GstcError("Failed to communicate with GSTD",
+                            GstcErrorCode.GSTC_UNREACHABLE)\
+                from e
+        except TypeError as e:
+            raise GstcError('Gst Client Bad command',
```

Comment:
`GstClient` bad command. Please double check all your strings. Our server name is `Gstd` and our client name is `GstClient`. Also, double check we don't have strange upper case words in the middle of the sentence (like `Bad` in this one).